app/controllers/ops_controller/settings/common.rb

```diff
@@ -247,6 +240,27 @@ def pglogical_validate_subscription
 
   private
 
+  PASSWOD_MASK = '●●●●●●●●'.freeze
+
+  def find_subscription(id = nil)
+    id.nil? ? PglogicalSubscription.new : PglogicalSubscription.find(id)
+  end
+
+  def set_subscription_attributes(subscription, params)
+    params_for_connection_validation(params).each do |k, v|
+      subscription.send("#{k}=".to_sym, v)
+    end
+  end
+
+  def params_for_connection_validation(subscription_params)
+    {'host'     => subscription_params[:host],
+     'port'     => subscription_params[:port],
+     'user'     => subscription_params[:user],
+     'dbname'   => subscription_params[:dbname],
+     'password' => subscription_params['password'] == PASSWOD_MASK ? nil : subscription_params['password']
```

Comment:
Why would subscription_params have some keys strings and others symbols?

It looks like they should all be strings if this logic was taken from [here](https://github.com/ManageIQ/manageiq/pull/10711/files#diff-84dbfcd1a7ac9f7442c66dadf41485bbL195)